electricitymap/contrib/lib/models/event_lists.py

```diff
@@ -189,6 +198,32 @@ def merge_exchanges(
 
         return exchanges
 
+    @staticmethod
+    def update_exchanges(
+        exchanges: "ExchangeList", new_exchanges: "ExchangeList", logger: Logger
+    ) -> "ExchangeList":
+        """Given a new batch of exchanges, update the existing ones."""
+        if len(new_exchanges) == 0:
+            return exchanges
+        elif len(exchanges) == 0:
+            return new_exchanges
+
+        for new_event in new_exchanges.events:
+            if new_event.datetime in exchanges:
+                existing_event = exchanges[new_event.datetime]
+                updated_event = Exchange.update(existing_event, new_event)
+                exchanges[new_event.datetime] = updated_event
+            else:
+                exchanges.append(
+                    new_event.zoneKey,
+                    new_event.datetime,
+                    new_event.source,
+                    new_event.netFlow,
+                    new_event.sourceType,
+                )
```

Comment:
With the changes in the follow up PR it will look like this instead:
```python
        for new_event in new_exchanges:
            if new_event in exchanges:
                existing_event = exchanges[new_event]
                updated_event = Exchange.update(existing_event, new_event)
                exchanges[new_event] = updated_event
            else:
                exchanges._append_event(new_event)
```
Which could be further reduced by removing the intermediate variables if we want.

Reply:
append is not a magic method in it self and to avoid too much logic in the magic methods this made the most sense imo, but I have made follow up changes locally.